providers/mlx5/mlx5.c

```diff
@@ -609,16 +610,35 @@ static int mlx5_map_internal_clock(struct mlx5_device *mdev,
 	return 0;
 }
 
+struct ibv_cq_ex *mlx5dv_create_cq_ex(struct ibv_context *context,
+				      struct ibv_cq_init_attr_ex *cq_attr,
+				      struct mlx5dv_cq_init_attr_ex *mlx5_cq_attr)
+{
+	return ibv_vendor_create_cq_ex(context, cq_attr, mlx5_cq_attr);
+}
```

Comment:
Basically I'm fine with some helpers, however it's not only a post helper. Why the lock that is taken in libibverbs before calling the provider is not needed here ? (i.e. pthread_mutex_lock(&context->mutex);) this locks currently comes to protect from parallel call to ibv_destroy_comp_channel which is still applicable here and in the future may be used for other purposes. If you think that having 2 helpers is better than calling one API that will callback the provider I can go with that as well.